Adding optional headers to handle rfc7386 required for some API calls

carsso opened the pull request:
Adding optional headers to handle rfc7386 (JSON merge patch through HttpHeaders.IF_MATCH header) required for some API calls (like `/organizations/{organizationUuid}/products/v2/{productUuid}`)

Can be used like this
`$guzzleIzettleClient->put($url, $data, ['If-Match' => '*']);`

carsso committed:
Adding optional headers to handle rfc7386 (JSON merge patch through HttpHeaders.IF_MATCH header) required for some API calls (like /organizations/{organizationUuid}/products/v2/{productUuid})

carsso committed:
Handle errors with no violations field

LauLaman commented:
Hi @carsso

Great work! can you fix the broken tests? so we can get this merged

carsso commented:
Hi @LauLaman

Sorry, I completely forgot about this PR. I bumped guzzle for one of my projects.

Newer guzzle version specifies return types in functions signatures.
So when mocking we have to go ont step further to match return types, that's why it was failing.

Tests are now fixed (and passing for guzzle 6.3 and 7.3). I also fixed the 2 small failures on uuid since #48 

I never used scrutinizer before so I don't know how to ask it to test both versions to make sur we don't break guzzle 6.3 in the future.

carsso commented:
@LauLaman Hi, is there any chance to get this PR merged ? Thanks

carsso committed:
Fixing getJson test in GuzzleIzettleClientTest

carsso committed:
Fixing composer run format

carsso commented:
hello @LauLaman 
Is is possible to merge that PR ? 
I rebasesd on master, fixed the conflicts, phpunit and phpcs : See https://github.com/carsso/iZettleApi/actions/runs/10408737052

carsso committed:
Updating Scrutinizer and Travis PHP versions as the project now requires php ^8.1
